websight-cms/templates/proxy/proxy-deployment.yaml

```diff
@@ -0,0 +1,88 @@
+{{- if .Values.proxy.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "websight-cms.component.fullname" (dict "componentName" "proxy" "context" $) }}
+  labels:
+    {{- include "websight-cms.component.labels" (dict "componentName" "proxy" "context" $) | nindent 4 }}
+spec:
+  replicas: {{ .Values.proxy.replicas }}
+  selector:
+    matchLabels:
+      {{- include "websight-cms.component.selectorLabels" (dict "componentName" "proxy" "context" $) | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "websight-cms.component.selectorLabels" (dict "componentName" "proxy" "context" $) | nindent 8 }}
+    spec:
+      {{- if .Values.proxy.imagePullSecrets }}
+      imagePullSecrets: {{- .Values.proxy.imagePullSecrets | toYaml | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: nginx
+          image: "{{ .Values.proxy.image.repository }}:{{ .Values.proxy.image.tag }}"
+          imagePullPolicy: {{ .Values.proxy.image.pullPolicy }}
+          env:
+            - name: NGINX_ENVSUBST_OUTPUT_DIR
+              value: /tmp/nginx/conf.d
+            {{- range $k, $v := .Values.proxy.env }}
+            - name: {{ $v.name }}
+              value: {{ $v.value | quote }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 80
+          {{- if .Values.proxy.resources }}
+          resources:
+{{ toYaml .Values.proxy.resources | indent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: tmp-config
+              mountPath: /tmp/nginx/conf.d
+            - name: nginx-config
+              mountPath: "/etc/nginx/conf.d"
+              readOnly: true
+            - name: nginx-config-templates
+              mountPath: "/etc/nginx/templates"
+              readOnly: true
+          {{- if .Values.proxy.livenessProbe.enabled }}
+          livenessProbe:
```

Comment:
We need to specify readiness probe here to prevent unstable nginx to be used in service routing (e.g during upgrades and invalid configs).